File: crates/source-languages/tests/fixtures/sample.java
package com.example.catalog;

import java.util.HashMap;
import java.util.Map;

public class Catalog implements Restockable {
    public static final int DEFAULT_PAGE_SIZE = 25;

    private final Map<String, Integer> items = new HashMap<>();

    public int add(String sku, int units) {
        String normalized = normalize(sku);
        int current = items.getOrDefault(normalized, 0);
        items.put(normalized, current + units);
        return current + units;
    }

    private static String normalize(String sku) {
        return sku.trim().toUpperCase();
    }

    @Override
    public void restock(int units) {
        add("DEFAULT", units);
    }

    public static Catalog buildCatalog() {
        Catalog catalog = new Catalog();
        catalog.add("a-1", DEFAULT_PAGE_SIZE);
        return catalog;
    }
}

File: crates/source-languages/tests/fixtures/sample.js
const DEFAULT_PAGE_SIZE = 25;

class Catalog {
  constructor() {
    this.items = new Map();
  }

  add(sku, units) {
    const normalized = Catalog.normalize(sku);
    const current = this.items.get(normalized) ?? 0;
    this.items.set(normalized, current + units);
    return current + units;
  }

  static normalize(sku) {
    return sku.trim().toUpperCase();
  }
}

function buildCatalog() {
  const catalog = new Catalog();
  catalog.add("a-1", DEFAULT_PAGE_SIZE);
  return catalog;
}

module.exports = { Catalog, buildCatalog };
